lib/tasks/taxonomy_event_log.rb

```diff
@@ -0,0 +1,59 @@
+class TaxonomyEventLog
+  def export
+    exported = []
+    previous_events = {}
+
+    raw_events.each do |event|
+      previous_taxons = previous_events[event.content_id].to_a
+      current_taxons = event.payload[:links][:taxons]
+
+      # Nil means that this `PatchLinkSet` doesn't change the `taxons` (which is
+      # different from sending an empty array)
+      next if current_taxons.nil?
+
+      (current_taxons - previous_taxons).each do |taxon_id|
+        exported << build_line(event, taxon_id, 1)
+      end
+
+      (previous_taxons - current_taxons).each do |taxon_id|
+        exported << build_line(event, taxon_id, -1)
+      end
+
+      previous_events[event.content_id] = current_taxons
+    end
+
+    exported
+  end
+
+  def build_line(event, taxon_id, change)
+    {
+      taggable_content_id: event.content_id,
+      taggable_title: fetch_title(event.content_id),
+      taxon_content_id: taxon_id,
+      taxon_title: fetch_title(taxon_id),
+      tagged_at: event.created_at,
+      tagged_on: event.created_at.to_date,
+      user_uid: event.user_uid,
+      change: change,
+    }
+  end
+
+private
+
+  def fetch_title(content_id)
+    @title_cache ||= {}
+
+    @title_cache[content_id] ||= begin
+      document = Document.find_by(content_id: content_id) || raise("#{content_id} not found")
+      latest_edition = document.editions.last
+      latest_edition ? latest_edition.title : "[Discarded draft]"
+    end
+  end
+
+  def raw_events
+    Event
+      .where('payload IS NOT NULL')
```

Comment:
I think it's best to use double quotes to stick with `publishing-api` conventions.